contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.2.0--4.3.0.sql

```diff
@@ -45,6 +45,106 @@ ALTER TABLE sys.babelfish_function_ext DROP CONSTRAINT babelfish_function_ext_pk
+		out_remarks sys.varchar(254)
+	)
+	AS $$
+		DECLARE opt_table sys.varchar(16) = '';
+		DECLARE opt_view sys.varchar(16) = '';
+		DECLARE cs_as_in_table_type varchar COLLATE "C" = in_table_type;
+	BEGIN
+		IF upper(cs_as_in_table_type) LIKE '%''TABLE''%' THEN
+			opt_table = 'TABLE';
+		END IF;
+		IF upper(cs_as_in_table_type) LIKE '%''VIEW''%' THEN
+			opt_view = 'VIEW';
+		END IF;
+		IF in_fusepattern = 1 THEN
+			RETURN query
+			SELECT 
+			CAST(table_qualifier AS sys.sysname) AS TABLE_QUALIFIER,
+			CAST(table_owner AS sys.sysname) AS TABLE_OWNER,
+			CAST(table_name AS sys.sysname) AS TABLE_NAME,
+			CAST(table_type AS sys.varchar(32)) AS TABLE_TYPE,
+			CAST(remarks AS sys.varchar(254)) AS REMARKS
+			FROM sys.sp_tables_view
+			WHERE (in_table_name IS NULL OR table_name LIKE in_table_name collate sys.database_default)
+			AND (in_table_owner IS NULL OR table_owner LIKE in_table_owner collate sys.database_default)
+			AND (in_table_qualifier IS NULL OR table_qualifier LIKE in_table_qualifier collate sys.database_default)
+			AND (cs_as_in_table_type IS NULL OR table_type = opt_table OR table_type = opt_view)
+			ORDER BY table_qualifier, table_owner, table_name;
+		ELSE 
+			RETURN query
+			SELECT 
+			CAST(table_qualifier AS sys.sysname) AS TABLE_QUALIFIER,
+			CAST(table_owner AS sys.sysname) AS TABLE_OWNER,
+			CAST(table_name AS sys.sysname) AS TABLE_NAME,
+			CAST(table_type AS sys.varchar(32)) AS TABLE_TYPE,
+			CAST(remarks AS sys.varchar(254)) AS REMARKS
+			FROM sys.sp_tables_view
+			WHERE (in_table_name IS NULL OR table_name = in_table_name collate sys.database_default)
+			AND (in_table_owner IS NULL OR table_owner = in_table_owner collate sys.database_default)
+			AND (in_table_qualifier IS NULL OR table_qualifier = in_table_qualifier collate sys.database_default)
+			AND (cs_as_in_table_type IS NULL OR table_type = opt_table OR table_type = opt_view)
+			ORDER BY table_qualifier, table_owner, table_name;
+		END IF;
+	END;
+$$
+LANGUAGE plpgsql STABLE;
+
+CREATE OR REPLACE PROCEDURE sys.sp_tables (
+    "@table_name" sys.nvarchar(384) = NULL,
+    "@table_owner" sys.nvarchar(384) = NULL, 
+    "@table_qualifier" sys.sysname = NULL,
+    "@table_type" sys.nvarchar(100) = NULL,
+    "@fusepattern" sys.bit = '1')
+AS $$
+	DECLARE @opt_table sys.varchar(16) = '';
+	DECLARE @opt_view sys.varchar(16) = '';
+BEGIN
+
+	-- Handle special case: Enumerate all databases when no filters are applied
+	IF (@table_qualifier = '%' AND @table_owner = '' AND @table_name = '')
+	BEGIN
+		-- Enumerate all databases when owner and name are blank
+		SELECT
+			CAST(d.name AS sys.sysname) AS TABLE_QUALIFIER,
+			CAST(NULL AS sys.sysname) AS TABLE_OWNER,
+			CAST(NULL AS sys.sysname) AS TABLE_NAME,
+			CAST(NULL AS sys.varchar(32)) AS TABLE_TYPE,
+			CAST(NULL AS sys.varchar(254)) AS REMARKS
+		FROM sys.databases d ORDER BY TABLE_QUALIFIER;
+		
+		RETURN;
+	END;
+
+	IF (@table_qualifier != '' AND LOWER(@table_qualifier) != LOWER(sys.db_name()))
```

Comment:
does this account for trailing spaces? Eg. `@tabled_qualifier = 'master        '`

Reply:
Yes, it will throw error in trailing as well as leading spaces case.